apps/aurora-portal/src/server/sessionFromToken.ts

```diff
@@ -0,0 +1,15 @@
+import { FastifyRequest, FastifyReply } from "fastify"
+import { SessionCookie } from "./sessionCookie"
+
+export async function restoreSessionFromToken(req: FastifyRequest, res: FastifyReply) {
+  // This endpoint receives a auth token, validates it and sets the session cookie
+  const { authToken, redirectUrl = "/" } = req.body as { authToken: string; redirectUrl?: string }
```

Comment:
Type assertion with `as` can lead to runtime errors if req.body is not what you expect... Why no use Zod?

Reply:
While Zod could be used for validation, I'm opting to maintain consistent error responses, as implemented in lines 8 and 9, rather than introducing Zod error handling at this point.